docs/tutorial-part-2.rst

```diff
@@ -0,0 +1,977 @@
+        "echo 'create_main_cluster = false' | sudo tee -a /etc/postgresql-common/createcluster.conf" \
+        "sudo apt-get install -q -y postgresql-11-auto-failover-1.4" \
+        "sudo usermod -a -G postgres ha-admin"
+
+  for node in paris-a paris-b paris-c paris-app amsterdam-a amsterdam-b amsterdam-c dublin-a
+  do
+  az vm run-command invoke \
+     --resource-group ha-demo-dim-$(echo $node | cut -f1 -d-) \
+     --name ha-demo-dim-${node} \
+     --command-id RunShellScript \
+     --scripts \
+        "curl https://install.citusdata.com/community/deb.sh | sudo bash" \
+        "sudo apt-get install -q -y postgresql-common" \
+        "echo 'create_main_cluster = false' | sudo tee -a /etc/postgresql-common/createcluster.conf" \
+        "sudo apt-get install -q -y postgresql-11-auto-failover-1.4" \
+        "sudo usermod -a -G postgres ha-admin" &
+  done
+  wait
+
+Run a monitor
+-------------
+
+The pg_auto_failover monitor is the first component to run. It periodically
+attempts to contact the other nodes and watches their health. It also
+maintains global state that “keepers” on each node consult to determine their
+own roles in the system.
+
+.. code-block:: bash
+
+   # on the monitor virtual machine
+
+   ssh -l ha-admin `azip monitor` -- \
+     pg_autoctl create monitor \
+       --auth trust \
+       --ssl-self-signed \
+       --pgdata monitor \
+       --pgctl /usr/lib/postgresql/11/bin/pg_ctl
+
+This command initializes a PostgreSQL cluster at the location pointed
+by the ``--pgdata`` option. When ``--pgdata`` is omitted, ``pg_autoctl``
+attempts to use the ``PGDATA`` environment variable. If a PostgreSQL
+instance had already existing in the destination directory, this command
+would have configured it to serve as a monitor.
+
+``pg_auto_failover``, installs the ``pgautofailover`` Postgres extension, and
+grants access to a new ``autoctl_node`` user.
+
+In the Quick Start we use ``--auth trust`` to avoid complex security settings.
+The Postgres `trust authentication method`__ is not considered a reasonable
+choice for production environments. Consider either using the ``--skip-pg-hba``
+option or ``--auth scram-sha-256`` and then setting up passwords yourself.
+
+__ https://www.postgresql.org/docs/current/auth-trust.html_
+
+At this point the monitor is created. Now we'll install it as a service with
+systemd so that it will resume if the VM restarts.
+
+.. code-block:: bash
+
+   ssh -l ha-admin `azip monitor` << CMD
+     pg_autoctl -q show systemd --pgdata ~ha-admin/monitor > pgautofailover.service
+     sudo mv pgautofailover.service /etc/systemd/system
+     sudo systemctl daemon-reload
+     sudo systemctl enable pgautofailover
+     sudo systemctl start pgautofailover
+   CMD
+
+
+Bring up the nodes
+------------------
+
+We’ll create the primary database using the ``pg_autoctl create`` subcommand.
+
+.. code-block:: bash
+
+   ssh -l ha-admin `azip paris-a` -- \
```

Comment:
Hmm, this one has trouble for me.

```
03:11:56 7590 INFO  Using default --ssl-mode "require"
03:11:56 7590 INFO  Using --ssl-self-signed: pg_autoctl will create self-signed certificates, allowing for encrypted network traffic
03:11:56 7590 WARN  Self-signed certificates provide protection against eavesdropping; this setup does NOT protect against Man-In-The-Middle attacks nor Impersonation attacks.
03:11:56 7590 WARN  See https://www.postgresql.org/docs/current/libpq-ssl.html for details
03:11:56 7590 INFO  Started pg_autoctl postgres service with pid 7593
03:11:56 7590 INFO  Started pg_autoctl node-init service with pid 7594
03:11:56 7593 INFO   /usr/bin/pg_autoctl do service postgres --pgdata ha -v
03:11:56 7594 WARN  Failed to connect to "postgres://autoctl_node@10.11.1.4/pg_auto_failover?sslmode=require", retrying until the server is ready
03:11:56 7594 WARN  The server at "postgres://autoctl_node@10.11.1.4/pg_auto_failover?sslmode=require" could not be contacted after 2 attempts in 0 seconds. This might indicate that the server is not running, or that there is something wrong with the given connection parameters (for example, wrong port number), or that there is a network connectivity problem (for example, a firewall blocking the connection request).
03:12:27 7594 WARN  The server at "postgres://autoctl_node@10.11.1.4/pg_auto_failover?sslmode=require" could not be contacted after 26 attempts in 31 seconds. This might indicate that the server is not running, or that there is something wrong with the given connection parameters (for example, wrong port number), or that there is a network connectivity problem (for example, a firewall blocking the connection request).
03:12:59 7594 WARN  The server at "postgres://autoctl_node@10.11.1.4/pg_auto_failover?sslmode=require" could not be contacted after 47 attempts in 63 seconds. This might indicate that the server is not running, or that there is something wrong with the given connection parameters (for example, wrong port number), or that there is a network connectivity problem (for example, a firewall blocking the connection request).
03:13:30 7594 WARN  The server at "postgres://autoctl_node@10.11.1.4/pg_auto_failover?sslmode=require" could not be contacted after 124 attempts in 94 seconds. This might indicate that the server is not running, or that there is something wrong with the given connection parameters (for example, wrong port number), or that there is a network connectivity problem (for example, a firewall blocking the connection request).
03:14:01 7594 WARN  The server at "postgres://autoctl_node@10.11.1.4/pg_auto_failover?sslmode=require" could not be contacted after 181 attempts in 125 seconds. This might indicate that the server is not running, or that there is something wrong with the given connection parameters (for example, wrong port number), or that there is a network connectivity problem (for example, a firewall blocking the connection request).
```

Meanwhile, on the monitor:

```
$ systemctl status pgautofailover
● pgautofailover.service - pg_auto_failover
   Loaded: loaded (/etc/systemd/system/pgautofailover.service; enabled; vendor preset: enabled)
   Active: active (running) since Fri 2020-10-02 03:11:28 UTC; 6min ago
 Main PID: 7697 (pg_autoctl)
    Tasks: 15 (limit: 4113)
   Memory: 28.5M
   CGroup: /system.slice/pgautofailover.service
           ├─7697 /usr/bin/pg_autoctl run
           ├─7700 pg_autoctl: start/stop postgres
           ├─7701 pg_autoctl: monitor listener
           ├─7709 /usr/lib/postgresql/11/bin/postgres -D /home/ha-admin/monitor -p 5432 -h *
           ├─7710 postgres: pg_auto_failover monitor: logger
           ├─7712 postgres: pg_auto_failover monitor: checkpointer
           ├─7713 postgres: pg_auto_failover monitor: background writer
           ├─7714 postgres: pg_auto_failover monitor: walwriter
           ├─7715 postgres: pg_auto_failover monitor: autovacuum launcher
           ├─7716 postgres: pg_auto_failover monitor: stats collector
           ├─7717 postgres: pg_auto_failover monitor: pg_auto_failover monitor
           ├─7718 postgres: pg_auto_failover monitor: logical replication launcher
           ├─7719 postgres: pg_auto_failover monitor: pg_auto_failover monitor worker
           ├─7720 postgres: pg_auto_failover monitor: pg_auto_failover monitor worker
           └─7721 postgres: pg_auto_failover monitor: autoctl_node pg_auto_failover [local] idle

$ pg_autoctl show state --pgdata monitor

Name |  Node |  Host:Port | LSN | Reachable |       Current State |      Assigned State
-----+-------+------------+-----+-----------+---------------------+--------------------
```

Reply:
@dimitri I tried this again today to check if it was intermittent. However I hit the same problem.